src/js/game/systems/item_processor.js

```diff
@@ -262,13 +255,6 @@ export class ItemProcessorSystem extends GameSystemWithFilter {
 
         // First, take items
         const items = processorComp.inputSlots;
-        processorComp.inputSlots = [];
```

Comment:
Removing this involves assuming the item processor won't try to take any input during the handler and signal processing, which is true currently but could be a limitation on future behavior. I don't think this change in particular has a large impact on performance.